packages/@aws-cdk/aws-apigateway/lib/usage-plan.ts

```diff
@@ -178,7 +178,9 @@ export class UsagePlan extends Resource {
    * @param apiKey
    */
   public addApiKey(apiKey: IApiKey): void {
-    new CfnUsagePlanKey(this, 'UsagePlanKeyResource', {
+    const prefix = 'UsagePlanKeyResource';
+    const id = this.node.tryFindChild(prefix) ? `${prefix}:${apiKey.node.id}` : prefix;
```

Comment:
- use `node.uniqueId` (`node.id` is not good enough)
- add a little comment explaining why you are only postfixing with the node ID in the 2nd child.

Reply:
> use node.uniqueId (node.id is not good enough)

Please see my response to your previous comment on this asking why not.

Reply:
The api key is not defined in your scope and there is nothing that prevents users from passing in two different apiKeys with the same id defined in two different parent scopes.